src/server.c

```diff
@@ -3951,18 +3951,32 @@ int processCommand(client *c) {
 
     /* Check if the user can run this command according to the current
      * ACLs. */
-    int acl_keypos;
-    int acl_retval = ACLCheckCommandPerm(c,&acl_keypos);
+    int acl_errpos;
+    int acl_retval = ACLCheckCommandPerm(c,&acl_errpos);
+    if (acl_retval == ACL_OK && c->cmd->proc == publishCommand)
```

Comment:
I don't think we need the check for ACLCheckPubsubPerm within publishCommand. We always specifically check for it before the function calls.

Reply:
Yes，I think so. I used to wonder whether I should remove this, but afraid do things wrong.